pkg/env/readers.go

```diff
@@ -12,14 +12,52 @@ import (
 	"os"
 	"os/user"
 	"path/filepath"
+	"sync"
 
 	"github.com/getoutreach/gobox/pkg/app"
 	"github.com/getoutreach/gobox/pkg/cfg"
 	"gopkg.in/yaml.v3"
 )
 
+type testOverrides struct {
```

Comment:
Not seeing the value of this over `sync.Map` at this point. Thoughts on using that?

Reply:
Also does not fix the race condition of two tests adding to that map at the same time.

Reply:
sync.Map isn't structured and they actually warn you against using it in the godoc for it unless its for very specific things.

Reply:
Fine w/ me, just seemed super duplicated. I get the type requirements here, but w/ just strings it didn't seem that beneficial